File: cpp/producer-consumer/producer-consumer/DataQueue.cpp
#include <memory>

#include "ChunkData.h"
#include "DataQueue.h"

namespace producer_consumer
{

	DataQueue::DataQueue()
	{
	}

	DataQueue::~DataQueue()
	{
	}

	void DataQueue::Inqueue(shared_ptr<ChunkData> data)
	{
		queue.push(data);
	}

	shared_ptr<ChunkData> DataQueue::Dequeue()
	{
		shared_ptr<ChunkData> data = queue.front();
		queue.pop();
		return data;
	}

	size_t DataQueue::Size() const
	{
		return queue.size();
	}

}  // namespace producer_consumer

File: java/producer-consumer/src/main/java/producer_consumer/DataQueue.java
package producer_consumer;

import java.util.LinkedList;
import java.util.Queue;

public class DataQueue {

    private Queue<ChunkData> queue;

    public DataQueue() {
        this.queue = new LinkedList<ChunkData>();
    }

    public void inqueue(ChunkData data) {
        queue.add(data);
    }

    public ChunkData dequeue() {
        return queue.remove();
    }
}
